request_token/models.py

```diff
@@ -238,7 +238,7 @@ def validate_max_uses(self):
 
     def _auth_is_anonymous(self, request):
         """Authenticate anonymous requests."""
-        assert request.user.is_anonymous(), 'User is authenticated.'
+        assert request.user.is_anonymous, 'User is authenticated.'
```

Comment:
Ah, this is bad. `assert` can be stripped by the compiler so this code could effectively be a `pass` on those that do so. We don't run with compiler optimisations stripping the asserts, but as this is OS I feel we have a little duty to fix this up.

Can be replaced with a conditional instead, which never get stripped.

More: https://dbader.org/blog/python-assert-tutorial

Reply:
True - but not part of this PR.